src/tools/debugger/session/Breakpoints.java

```diff
@@ -91,20 +108,42 @@ public Breakpoint addOrUpdate(final AsyncMessageReceiveBreakpoint bId) {
       truffleBreakpoints.put(bId, bp);
     }
     bp.setEnabled(bId.isEnabled());
-    return bp;
   }
 
-  public synchronized void addOrUpdate(final MessageReceiveBreakpoint bId) {
+  public synchronized void addOrUpdate(final MessageReceiverBreakpoint bId) {
     FullSourceCoordinate coord = bId.getCoordinate();
-    BreakpointEnabling<MessageReceiveBreakpoint> existingBP = receiverBreakpoints.get(coord);
+    BreakpointEnabling<MessageReceiverBreakpoint> existingBP = receiverBreakpoints.get(coord);
     if (existingBP == null) {
-      existingBP = new BreakpointEnabling<MessageReceiveBreakpoint>(bId);
+      existingBP = new BreakpointEnabling<MessageReceiverBreakpoint>(bId);
       receiverBreakpoints.put(coord, existingBP);
     } else {
       existingBP.setEnabled(bId.isEnabled());
     }
   }
 
+  public synchronized void addOrUpdate(final PromiseResolverBreakpoint bId) {
```

Comment:
At this point you don't really need the info about `extends SectionBreakpoint` or do you?
Guess it would work if you drop that part?